src/routers/alpha-router/alpha-router.ts

```diff
@@ -1733,6 +1748,22 @@ export class AlphaRouter
     swapConfig?: SwapOptions,
     providerConfig?: ProviderConfig
   ): Promise<BestSwapRoute | null> {
+    const tokenPairProperties =
```

Comment:
nit: maybe lets make this a util function? seems like we call it a few times?

Reply:
I tried it, and somehow some unit tests in `alpha-router.test.ts` started failing.

Reply:
I suspect it's mocking the token properties in those unit tests at the alpha-router class, which makes it kinda hard to do this kind of refactoring.